pkg/sparrow/run_test.go

```diff
@@ -189,19 +190,29 @@ func Test_fanInResults(t *testing.T) {
 	close(checkChan)
 }
 
+// TestSparrow_Run tests that the Run method starts the API
 func TestSparrow_Run(t *testing.T) {
+	// create simple file loader config
 	c := &config.Config{
 		Api: config.ApiConfig{ListeningAddress: ":9090"},
 		Loader: config.LoaderConfig{
-			Type: "http",
+			Type:     "file",
+			Interval: time.Second * 1,
 		},
 	}
 
-	s := New(c)
+	c.SetLoaderFilePath("../config/testdata/config.yaml")
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-	if err := s.Run(ctx); err != nil && !errors.Is(err, context.DeadlineExceeded) {
-		t.Errorf("Sparrow.Run() error = %v", err)
-	}
```

Comment:
No idea why this test was passing, it almost always failed in my IDE. I've made the the test a positive case; the API starts and no error is thrown.